File: src-tauri/src/model/session.rs
use serde::{Deserialize, Serialize};

use crate::model::filter::FilterState;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NetworkInterface {
    pub name: String,
    pub description: String,
    pub addresses: Vec<String>,
    pub is_loopback: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CaptureSessionMeta {
    pub id: String,
    pub name: String,
    pub interface_name: String,
    pub started_at_ms: i64,
    pub ended_at_ms: Option<i64>,
    pub packet_count: u64,
    pub bytes_captured: u64,
    pub running: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CaptureRuntimeState {
    pub active_session_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TlsDecryptionConfig {
    pub enabled: bool,
    pub keylog_path: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StartCaptureRequest {
    pub interface_name: String,
    pub filter: Option<FilterState>,
}

File: src/types/session.ts
import type { FilterState } from "./filter";

export interface NetworkInterface {
  name: string
  description: string
  addresses: string[]
  is_loopback: boolean
}

export interface CaptureSessionMeta {
  id: string
  name: string
  interface_name: string
  started_at_ms: number
  ended_at_ms: number | null
  packet_count: number
  bytes_captured: number
  running: boolean
}

export interface CaptureRuntimeState {
  active_session_id: string | null
}

export interface TlsDecryptionConfig {
  enabled: boolean
  keylog_path: string | null
  reload_on_change: boolean
  strict_secret_match: boolean
}

export interface StartCaptureRequest {
  interface_name: string
  filter: FilterState | null
}
